### PR TITLE
feat: link_provider app

### PR DESCRIPTION
**Pull request by matekdev**

### CODE REVIEW

**Comment by matekdev** [comment]
@MisterJimson I created the new app and moved over the redirection logic. I think the last few parts would be...

1. I imagine we need to create a separate vercel project for this?
2. Maybe some way of passing down config data to the `link_provider` such as the database details, and the URL it should redirect to?

I'm going to leave this as is for now and work on other things as I'm unclear with the remaining details.

**Comment by MisterJimson** [comment]
1. Yes we would need another vercel project. I can set that up when we merge
2. Which URL would we redirect to? It should always come from the database or 404 as far as I can think of.

File: apps/link_provider/src/routes/+layout.server.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+			.from('dynamic_links')
+			.select('deep_link')
+			.eq('link', link)
+			.single();
+
+		if (!linkError && linkData) throw redirect(303, linkData.deep_link);
+		else throw error(404, 'It looks like that link no longer exists...');
+	}
+
+	throw redirect(303, redirectLink);
```
[the remark this answers is not in the document]
Comment (matekdev) [reply apps/link_provider/src/routes/+layout.server.ts:23]: Right I see what you mean, thanks.

**Comment by matekdev** [comment]
Feedback addressed

**Comment by MisterJimson** [comment]
Ping me when this is merged and I'll set it up in Vercel. We may want to keep a reserved subdomain for our own testing that can't be removed. Like demo.turt.link.

**Pull request merged**